app/src/main/java/com/github/Jenjamin3000/bootcamp/MapsFragment.java

```diff
@@ -0,0 +1,167 @@
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.Fragment;
+
+import android.Manifest;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.pm.PackageManager;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.github.Jenjamin3000.bootcamp.utils.PermissionUtils;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.maps.model.Polygon;
+import com.google.android.gms.maps.model.PolygonOptions;
+
+public class MapsFragment extends Fragment{
+
+    private GoogleMap map;
+    private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
+    /**
+     * Flag indicating whether a requested permission has been denied
+     */
+    private boolean permissionDenied = false;
+    private OnMapReadyCallback callback = googleMap -> {
+        map = googleMap;
+
+        enableMyLocation();
+
+        //Listener when user clicks on the "my position" button
+        map.setOnMyLocationButtonClickListener(() -> {
+            Toast.makeText(getActivity(), "MyLocation button clicked", Toast.LENGTH_SHORT).show();
+            return false;
+        });
+
+        //Listener when user clicks on the "my position" blue dot
+        map.setOnMyLocationClickListener(location -> {
+            Toast.makeText(getActivity(), "Current location:\n" + location, Toast.LENGTH_LONG).show();
+        });
+
+        // Add a marker at Satellite and move the camera
+        LatLng epfl = new LatLng(46.520536, 6.568318);
+        LatLng satellite = new LatLng(46.520544, 6.567825);
+        map.addMarker(new MarkerOptions().position(satellite).title("Satellite").snippet("5 ⭐"));
+
+        LatLngBounds epflBounds = new LatLngBounds(
+                new LatLng(46, 6), // SW bounds
+                new LatLng(47, 7)  // NE bounds
+        );
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(epfl, 15));
+
+        map.setOnMarkerClickListener(marker -> {
+            // Triggered when user click any marker on the map
+            // See example below:
+           /* Toast.makeText(getActivity(),
+                    marker.getTitle(),
+                    Toast.LENGTH_SHORT).show();
+
+            */
+            return false;
+        });
+
+        Polygon campus = map.addPolygon(new PolygonOptions()
+                .clickable(true)
+                .add(
+                        new LatLng(46.5223, 6.5633),
```

Comment:
We'll need a way to automate this... further down the line when we'll be splitting the map into zones, build a reference class which contains the entirety of the map, and especially the corners of all of the zones. Maybe build a LUT? A .dat file?

Reply:
Yes, utimately, the goal is to have files to store zones, I can work on this next week for the next issue